internal/worker/sshserver/server_test.go

```diff
@@ -154,6 +192,60 @@ func (s *sshServerSuite) TestSSHServer(c *gc.C) {
 	workertest.CleanKill(c, server)
 }
 
+func (s *sshServerSuite) TestHostKeyForTarget(c *gc.C) {
+	ctrl := gomock.NewController(c)
+	defer ctrl.Finish()
+	facadeClient := NewMockFacadeClient(ctrl)
+
+	facadeClient.EXPECT().HostKeyForTarget(gomock.Any()).Return(s.hostKey, nil)
+	// Firstly, start the server on an in-memory listener
+	listener := bufconn.Listen(8 * 1024)
+
+	_, err := sshserver.NewServerWorker(sshserver.ServerWorkerConfig{
+		Logger:               loggo.GetLogger("test"),
+		Listener:             listener,
+		JumpHostKey:          jujutesting.SSHServerHostKey,
+		NewSSHServerListener: newTestingSSHServerListener,
+		FacadeClient:         facadeClient,
+	})
+	c.Assert(err, jc.ErrorIsNil)
+	// Open a client connection
+	client := inMemoryDial(c, listener, &ssh.ClientConfig{
+		User:            "",
+		HostKeyCallback: ssh.InsecureIgnoreHostKey(),
+		Auth: []ssh.AuthMethod{
+			ssh.Password(""), // No password needed
+		},
+	})
+	conn, err := client.Dial("tcp", "localhost:0")
+	c.Assert(err, gc.IsNil)
+	// we need to establish another client connection to perform the auth in the embedded server.
+	_, _, _, err = ssh.NewClientConn(
+		conn,
+		"",
+		&ssh.ClientConfig{
+			HostKeyCallback: ssh.FixedHostKey(s.publicHostKey),
+			Auth: []ssh.AuthMethod{
+				ssh.PublicKeys(s.userSigner),
+			},
+		},
+	)
+	c.Assert(err, gc.IsNil)
+
+	// we now open a new connection but the facadeClient will return an error, we'll close the channel,
+	// making the ssh handshake to error out for the embedded server.
```

Comment:
What about just this
```suggestion
	// we now test that the connection is closed when the controller cannot fetch the unit's host key.
```

Reply:
Agreed